students/jstonham/README.md

```diff
@@ -2873,4 +2873,149 @@ Controller - Business logic.
 In MVC frameworks, the model and view are decoupled.
 This means they don't interact with each other and can therefore be updated independently of each other.
 
-Examples of MVC frameworks are Rails (written in Ruby) and Spring MVC (written in Java).
\ No newline at end of file
+Examples of MVC frameworks are Rails (written in Ruby) and Spring MVC (written in Java).
+
+# React
+
+React takes an entire web page and splits it into components.
+These components are usually a combination of JavaScript, HTML and CSS.
+
+The JavaScript part of React allows classes and encapsulation.
+This allows components to be kept separate and updated independently of each other.
+
+Part of the reason that React is really performant is that change can only happen in one direction.
+Unlike other tools such as Angular and Ember, where change can flow from a component's state to the user *or* from the user to a component's state, React *only* allows change to flow from a component's state to the user.
+
+React components are made up of tags.
+There can only be one outermost tag in a React component, but other tags can be nested inside.
+In the following example, the `render()` method has an outermost `div` tag with `h1` and `ul` tags nested inside, and inside the `ul` tag are 3 `li` tags:
+
+```JSX
+render() {
+        return (
+            <div className="shopping-list">
+            <h1>ShoppingList for {this.props.name}</h1>
+            <ul>
+                <li>Instagram</li>
+                <li>WhatsApp</li>
+                <li>Oculus</li>
+            </ul>
+            </div>
+        );
+    }
+  ```
+
+## JSX (JavaScript Extended)
+
+JavaScript Extended is JavaScript with HTML.
+React can take in JSX and render as HTML.
+
+The following is a React component called `Greeting`, written in JSX, that will render the greeting message in HTML.
+
+```JSX
+class Greeting extends React.Component {
+    render() {
+        const name = this.props.name
+        return <p> Hi! My name is {name} </p>
+    }
+}
+```
+
+React components only need one method, `render()`, which will return the new component in a JSX format.
+
+Due to the fact that it's a combination of JavaScript and HTML, JSX needs to be able to differentiate between its JavaScript and HTML classes.
+The way it does so is to call its JavaScript classes `class` and its HTML classes `className`.
+
+## Data Structure
+
+Computer scientists think of data structures as trees (although it's easier to think of them as root systems).
+
+Example data structure:
+
+| | | Root (main Parent) | | | |
```

Comment:
Tremendous effort on this bit, but maybe it's easier to use a picture here instead?